app/services/config_report_importer.rb

```diff
@@ -0,0 +1,21 @@
+class ConfigReportImporter < ReportImporter
+  attr_reader :report
+  def self.authorized_smart_proxy_features
+    ['Puppet']
+  end
+
+  def report_name_class
+    ConfigReport
+  end
+
+  private
+
+  def create_report_and_logs
+    @report = super
```

Comment:
I still can't see a use for setting this instance variable or the attr_reader above:

<pre>
From: /home/dcleal/code/foreman/foreman/app/services/config_report_importer.rb @ line 15 ConfigReportImporter#create_report_and_logs:

    13: def create_report_and_logs
    14:   super
 => 15:   binding.pry
    16:   return report unless report.persisted?
    17:   # Store all Puppet message logs
    18:   import_log_messages
    19:   # Check for errors
    20:   notify_on_report_error(:puppet_error_state)
    21: end

[1] pry(#<ConfigReportImporter>)> 
[2] pry(#<ConfigReportImporter>)> report
=> #<ConfigReport id: 1345, host_id: 109, reported_at: "2015-08-24 09:01:02", created_at: "2015-08-24 09:01:05", updated_at: "2015-08-24 09:01:05", status: 1, metrics: "--- !ruby/hash:ActiveSupport::HashWithIndifferentAc...", type: "ConfigReport">
</pre>